milestones/api.py

```diff
@@ -255,38 +272,59 @@ def remove_course_milestone(course_key, milestone):
     return data.delete_course_milestone(course_key=course_key, milestone=milestone)
 
 
-def add_course_content_milestone(course_key, content_key, relationship, milestone):
+def add_course_content_milestone(course_key, content_key, relationship, milestone, requirements=None):
     """
     Adds a course-content-milestone link to the system
-    'relationship': string value (eg: 'requires')
+
+    Arguments:
+        course_key (CourseKey|str): CourseKey of the course containing the content
+        content_key (UsageKey|str): UsageKey of the content
+        relationship (str): The type of relationship that the content shares with the milestone (e.g. 'requires')
+        milestone (dict): The milestone which the given content relates to
+        requirements (dict): Data that is used to determine user fulfillment of the milestone
+
+    Returns:
+        None
     """
     _validate_course_key(course_key)
     _validate_content_key(content_key)
     _validate_milestone_relationship_type(relationship)
     _validate_milestone_data(milestone)
+    _validate_requirements(requirements)
```

Comment:
Actually, I think we should leave the existing validators convention. Putting the validator on the model class would require us to reference the model class outside of the data module which is something that this package seems to want to avoid.